test/server/active_tcp_listener_test.cc

```diff
@@ -386,6 +407,111 @@ TEST_F(ActiveTcpListenerTest, ListenerFilterWithInspectDataMultipleFilters2) {
+        manager.addAcceptFilter(nullptr, Network::ListenerFilterPtr{no_inspect_data_filter2});
+        manager.addAcceptFilter(nullptr, Network::ListenerFilterPtr{inspect_data_filter3});
+        return true;
+      }));
+
+  auto listener = std::make_unique<NiceMock<Network::MockListener>>();
+  EXPECT_CALL(*listener, onDestroy());
+  Network::NopConnectionBalancerImpl balancer;
+  Network::Address::InstanceConstSharedPtr address(
+      new Network::Address::Ipv4Instance("127.0.0.1", 10001));
+  auto active_listener = std::make_unique<ActiveTcpListener>(
+      conn_handler_, std::move(listener), address, listener_config_, balancer, runtime_);
+  auto accepted_socket = std::make_unique<NiceMock<Network::MockConnectionSocket>>();
+
+  EXPECT_CALL(*accepted_socket, ioHandle()).WillRepeatedly(ReturnRef(io_handle_));
+  EXPECT_CALL(io_handle_, isOpen()).WillRepeatedly(Return(true));
+  Event::FileReadyCb file_event_callback;
+
+  EXPECT_CALL(io_handle_,
+              createFileEvent_(_, _, Event::PlatformDefaultTriggerType,
+                               Event::FileReadyType::Read | Event::FileReadyType::Closed))
+      .WillOnce(SaveArg<1>(&file_event_callback));
+  EXPECT_CALL(io_handle_, recv)
+      .WillOnce([&](void*, size_t size, int) {
+        EXPECT_EQ(128, size);
+        return Api::IoCallUint64Result(128, Api::IoError::none());
+      })
+      .WillOnce([&](void*, size_t size, int) {
+        EXPECT_EQ(512, size);
+        return Api::IoCallUint64Result(512, Api::IoError::none());
+      })
+      .WillOnce([&](void*, size_t size, int) {
+        EXPECT_EQ(512, size);
+        return Api::IoCallUint64Result(512, Api::IoError::none());
+      });
+
+  EXPECT_CALL(*no_inspect_data_filter1, onAccept(_))
+      .WillOnce(Return(Network::FilterStatus::StopIteration));
+
+  active_listener->incNumConnections();
+  // Calling the onAcceptWorker() to create the ActiveTcpSocket.
+  active_listener->onAcceptWorker(std::move(accepted_socket), false, true);
+
+  EXPECT_CALL(*inspect_data_filter1, onAccept(_))
+      .WillOnce(Return(Network::FilterStatus::StopIteration));
+  EXPECT_CALL(io_handle_, activateFileEvents(Event::FileReadyType::Read));
+
+  ActiveTcpSocket* tcp_socket1 = active_listener->sockets().front().get();
+  tcp_socket1->continueFilterChain(true);
+
+  EXPECT_CALL(*inspect_data_filter1, onData(_)).WillOnce(Return(Network::FilterStatus::Continue));
+  EXPECT_CALL(*inspect_data_filter2, onAccept(_))
+      .WillOnce(Return(Network::FilterStatus::StopIteration));
+  EXPECT_CALL(io_handle_, activateFileEvents(Event::FileReadyType::Read));
+
+  EXPECT_TRUE(file_event_callback(Event::FileReadyType::Read).ok());
+
+  EXPECT_CALL(*inspect_data_filter2, onData(_)).WillOnce(Return(Network::FilterStatus::Continue));
+  EXPECT_CALL(*no_inspect_data_filter2, onAccept(_))
+      .WillOnce(Return(Network::FilterStatus::StopIteration));
+
+  EXPECT_TRUE(file_event_callback(Event::FileReadyType::Read).ok());
```

Comment:
Is there anyway we can test the close event?

Reply:
Are you talking about multi-platform support? To be honest it's kind of like a dark corner which all current tests don't touch, and it may be done in io_socket_handle test.

Reply:
We just test the close event, like the way in this test, you can trigger that by file_event_callback(Event::FileReadyType::Close)

Reply:
Added test ListenerFilterWithClose2 for close event